coriolis/osmorphing/base.py

```diff
@@ -342,3 +343,28 @@ def _set_selinux_autorelabel(self):
                 "touch /.autorelabel")
         except Exception as err:
             LOG.warning("Failed to set autorelabel: %r" % err)
+
+    def _retain_user_credentials(self):
+        cloud_cfg_path = "/etc/cloud/cloud.cfg"
+        if not self._test_path(cloud_cfg_path):
+            LOG.warn("Could not find %s. Skipping reconfiguration." % (
+                cloud_cfg_path))
+            return
+
+        self._event_manager.progress_update(
+            "Reconfiguring cloud-init to retain original user credentials")
+        try:
+            self._exec_cmd_chroot(
+                "cp %s %s.bak" % (cloud_cfg_path, cloud_cfg_path))
+
+            cloud_cfg = yaml.load(self._read_file(cloud_cfg_path),
+                                  Loader=yaml.SafeLoader)
+
+            cloud_cfg['disable_root'] = 0
+            cloud_cfg['ssh_pwauth'] = 1
```

Comment:
I tested this with 'dvincze' user, and also a 'cloudbase' user. No password got reset. I'm thinking maybe because we enable password login though SSH, cloud-init will stop reseting them. I can check that also.